ngraph/src/ngraph/frontend/onnx_import/onnx.cpp

```diff
@@ -43,17 +43,50 @@ namespace ngraph
 
                 struct stream_parse : ngraph_error
                 {
-                    explicit stream_parse(std::istream&)
-                        : ngraph_error{"Failure parsing data from the provided input stream"}
+                    explicit stream_parse(const std::string& stream_type)
+                        : ngraph_error{"Failure parsing data from the provided input " +
+                                       stream_type + "."}
+                    {
+                    }
+                };
+
+                struct stream_corrupted : ngraph_error
+                {
+                    explicit stream_corrupted()
+                        : ngraph_error{"Provided input stream is in incorrect state."}
                     {
                     }
                 };
 
             } // namespace error
-        }     // namespace detail
+
+            std::shared_ptr<Function> get_ng_function(const ONNX_NAMESPACE::ModelProto& model_proto)
+            {
+                Model model{model_proto};
+                Graph graph{model_proto.graph(), model};
+                auto function = std::make_shared<Function>(
+                    graph.get_ng_outputs(), graph.get_ng_parameters(), graph.get_name());
+                for (std::size_t i{0}; i < function->get_output_size(); ++i)
+                {
+                    function->get_output_op(i)->set_friendly_name(
+                        graph.get_outputs().at(i).get_name());
+                }
+                return function;
+            }
+        } // namespace detail
 
         std::shared_ptr<Function> import_onnx_model(std::istream& stream)
         {
+            if (!stream.good())
+            {
+                stream.clear();
+                stream.seekg(0);
+                if (!stream.good())
+                {
+                    throw detail::error::stream_corrupted();
+                }
+            }
```

Comment:
why do you except an empty stream to be the good one?
```suggestion
            if (!stream.good())
            {
                    throw detail::error::stream_corrupted();
            }
```

Reply:
This won't work. I've tried this. I'm not sure yet why but it's probable that during some conversions to stream etc in OpenVINO the stream is rewind till the end. So this lets' to recover from such situation.